tests/Dockerfile

```diff
@@ -0,0 +1,21 @@
+# The Dockerfile's resulting image is purpose-built for executing Jaeger Operator e2e tests within the OpenShift release (https://github.com/openshift/release) using Prow CI.
+FROM golang:1.20
+
+# Copy the repository files
+COPY . /tmp/jaeger-operator
+
+WORKDIR /tmp
+
+# Install kuttl
+RUN curl -L -o kuttl https://github.com/kudobuilder/kuttl/releases/download/v0.15.0/kubectl-kuttl_0.15.0_linux_x86_64 \
+    && chmod +x kuttl \
+    && mv kuttl /usr/local/bin/kuttl
+
+# Install kubectl and oc
+RUN curl -L -o oc.tar.gz https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/latest/openshift-client-linux.tar.gz \
```

Comment:
Maybe you can use the scripts from this repo to install the tools. Doing that, we ensure we use the same version as other parts of the automation. And we don't have to update multiple sources when increasing the version of one of those tools

Reply:
I have updated the Dockerfile, checking if there are any issues encountered when running the tests. I've skipped building the assert image as we will be providing a pre-built image as env var inside the container.

Reply:
@iblancasa I ran the tests using image built from the Dockerfile and the tests are running fine. Can you help to get this merged. Thanks.